src/test/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannelIT.java

```diff
@@ -562,4 +563,168 @@ public void testPartialBatchChannelInvalidationIngestion_schematization() throws
+
+    InMemorySinkTaskContext inMemorySinkTaskContext =
+        new InMemorySinkTaskContext(Collections.singleton(topicPartition));
+
+    // This will automatically create a channel for topicPartition.
+    SnowflakeSinkService service =
+        SnowflakeSinkServiceFactory.builder(conn, IngestionMethodConfig.SNOWPIPE_STREAMING, config)
+            .setRecordNumber(1)
+            .setErrorReporter(new InMemoryKafkaRecordErrorReporter())
+            .setSinkTaskContext(inMemorySinkTaskContext)
+            .addTask(testTableName, topicPartition)
+            .build();
+
+    TopicPartitionChannel topicPartitionChannel =
+        ((SnowflakeSinkServiceV2) service)
+            .getTopicPartitionChannelFromCacheKey(testChannelName)
+            .get();
+    // Channel does exist
+    Assert.assertNotNull(topicPartitionChannel);
+
+    // get the corresponding V2 format for above topic partition channel
+    final String channelNameFormatV2 =
+        topicPartitionChannel.generateChannelNameFormatV2(testChannelName, conn.getConnectorName());
+
+    // create a channel with new format and ingest few rows
+    // Ctor of TopicPartitionChannel tries to open the channel (new format) for same partition
+    TopicPartitionChannel topicPartitionChannelForFormatV2 =
+        new TopicPartitionChannel(
+            ((SnowflakeSinkServiceV2) service).getStreamingIngestClient(),
+            topicPartition,
+            channelNameFormatV2,
+            testTableName,
+            new StreamingBufferThreshold(10, 10_000, 1),
+            config,
+            new InMemoryKafkaRecordErrorReporter(),
+            new InMemorySinkTaskContext(Collections.singleton(topicPartition)),
+            conn,
+            conn.getTelemetryClient());
+
+    // insert few records via new channel
+    final int noOfRecords = 5;
+    // Since record 0 was not able to ingest, all records in this batch will not be added into the
+    // buffer.
+    List<SinkRecord> records =
+        TestUtils.createJsonStringSinkRecords(0, noOfRecords, testTableName, PARTITION);
+
+    records.forEach(topicPartitionChannelForFormatV2::insertRecordToBuffer);
+    TestUtils.assertWithRetry(
+        () -> topicPartitionChannelForFormatV2.getOffsetSafeToCommitToKafka() == noOfRecords, 5, 5);
+
+    // we migrate the offset from new channel format to old channel format
+    ChannelMigrateOffsetTokenResponseDTO channelMigrateOffsetTokenResponseDTO =
+        conn.migrateStreamingChannelOffsetToken(
+            testTableName, channelNameFormatV2, testChannelName);
+    Assert.assertTrue(isChannelMigrationResponseSuccessful(channelMigrateOffsetTokenResponseDTO));
+    Assert.assertEquals(
+        SUCCESS.getStatusCode(), channelMigrateOffsetTokenResponseDTO.getResponseCode());
+
+    // Fetch offsetToken from API should now give you same as other channel
+    TestUtils.assertWithRetry(
+        () -> service.getOffset(new TopicPartition(topic, PARTITION)) == noOfRecords, 5, 5);
```

Comment:
ditto on the above - can we try ingestion after migration as well?

Reply:
Ahh i realized, the ingestion is happening in testChannelMigrateOffsetTokenSystemFunction_NullOffsetTokenInFormatV2 after migration.

Reply:
But no harm in adding it here, so I have added in recent commit!